cfgmgr/vxlanmgr.cpp

```diff
@@ -600,22 +1052,170 @@ void VxlanMgr::clearAllVxlanDevices()
+    /* Currently, there is only 1 src vtep tunnel, hence picking the src_ip from that entry */
+    Table vxlanAppTunnelTable = Table(this->m_app_db, APP_VXLAN_TUNNEL_TABLE_NAME);
+    vector<std::string> appVxlanTunnelTableKeys;
+    vxlanAppTunnelTable.getKeys(appVxlanTunnelTableKeys);
+    std::string src_ip;
+    std::string dst_ip("");
+    dst_ip = "";
+    for (auto &k : appVxlanTunnelTableKeys)
+    {
+        std::vector<FieldValueTuple> temp;
+        if (vxlanAppTunnelTable.get(k, temp))
         {
-            info.m_vxlan = device_name;
-            cmdDeleteVxlan(info, res);
+            for (auto fv: temp)
+            {
+                std::string field = fvField(fv);
+                std::string value = fvValue(fv);
+                SWSS_LOG_NOTICE("RESTORE Vxlan Tunnel Table key: %s field: %s value: %s",
+                        k.c_str(), field.c_str(), value.c_str());
+                if (field == "src_ip")
+                {
+                    src_ip = value;
+                    SWSS_LOG_NOTICE("RESTORE Vxlan Tunnel Table src_ip: %s", src_ip.c_str());
+                }
+            }
         }
-        else if (device_name.find(VXLAN_IF_NAME_PREFIX) == 0)
+        else
         {
-            info.m_vxlanIf = device_name;
-            cmdDeleteVxlanIf(info, res);
+            SWSS_LOG_NOTICE("RESTORE VxLAN Tunnel Table Key(%s)", k.c_str());
         }
     }
+
+    Table vxlanAppTunnelMapTable = Table(this->m_app_db, APP_VXLAN_TUNNEL_MAP_TABLE_NAME);
+    std::vector<std::string>::iterator it;
+    for (it = m_appVxlanTunnelMapKeysRecon.begin();
+         it != m_appVxlanTunnelMapKeysRecon.end();
+         it++)
+    {
+        std::string vlan, vlan_id, vni_id;
+        std::string vxlanTunnelMapName = *it;
+        std::vector<FieldValueTuple> temp; 
+        if (vxlanAppTunnelMapTable.get(vxlanTunnelMapName, temp))
+        {
+            for (auto fv: temp)
+            {
+                std::string field = fvField(fv);
+                std::string value = fvValue(fv);
+                SWSS_LOG_NOTICE("RESTORE Vxlan Tunnel MAP Table key: %s field: %s value: %s",
+                        vxlanTunnelMapName.c_str(), field.c_str(), value.c_str());
+                if (field == VLAN)
+                {
+                    vlan = value;
+                }
+                else if (field == VNI)
+                {
+                    vni_id = value;
+                }
+            }
+        }
+        else
+        {
+            SWSS_LOG_NOTICE("RESTORE VxLAN Tunnel Map Table Key(%s)", vxlanTunnelMapName.c_str());
+        }
+        const auto vlan_prefix = std::string("Vlan");
+        const auto prefix_len = vlan_prefix.length();
+        vlan_id = vlan.substr(prefix_len);
+
+        size_t found = vxlanTunnelMapName.find(delimiter);
+        const auto vxlanTunnelName = vxlanTunnelMapName.substr(0, found);
+
+        createVxlanNetdevice(vxlanTunnelName, vni_id, src_ip, dst_ip, vlan_id);
```

Comment:
in this, it wouldn't be creating the netdevice right, since `m_in_reconcile` will be true? Just to clarify, will discuss in the review meeting

Reply:
Yes, this is correct.
During the initialization, vxlanmgrd reads the kernel and creates a cache of all the existing new devices.
When the configuration is restored, if the cache contains the corresponding kernel device, it will skip to create the kernel netdevice creation.
Typically for the swss docker restart, the cache will contain all the kernel netdevice and hence will not try to recreate it.
However for system warmboot, the kernel cache will be empty and hence while restoring the configuration, the corresponding kernel netdevices are created.

The m_in_reconcile is set to true (beginReconcile) before the configuration is restored.